narwhal/types/src/primary.rs

```diff
@@ -154,9 +155,72 @@ impl Hash<{ crypto::DIGEST_LENGTH }> for Batch {
     }
 }
 
+#[derive(Clone, Deserialize, MallocSizeOf, Serialize)]
+#[enum_dispatch(HeaderAPI)]
+pub enum Header {
+    V1(HeaderV1),
+}
+
+// TODO(arun): Is this default useful?
+impl Default for Header {
+    fn default() -> Self {
+        Self::V1(HeaderV1::default())
+    }
+}
+
+impl Header {
+    // TODO(arun): Should I be passing in a version number here and matching on that?
```

Comment:
@mystenmark ^^

Reply:
We can pass in a version number eventually. But this is fine for now.